### PR TITLE
feat: Course Outline  - Sections list

### PR DESCRIPTION
**Pull request by vladislavkeblysh**
- Added CardHeader component
- Added SectionCard component

[2u-336](https://youtrack.raccoongang.com/issue/2U-336)

### CODE REVIEW

File: src/course-outline/card-header/CardHeader.jsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+      </OverlayTrigger>
+      <Dropdown data-testid="section-card-header__menu" className="ml-auto">
+        <Dropdown.Toggle
+          className="section-card-header__menu"
+          id="section-card-header__menu"
+          as={IconButton}
+          src={MoveVertIcon}
+          alt="section-card-header__menu"
+          iconAs={Icon}
+          variant="primary"
```
Comment (PKulkoRaccoonGang) [inline src/course-outline/card-header/CardHeader.jsx:76]: By default Paragon Dropdown.Toggle component have a `primary` variant. Please remove this declaration.

File: src/course-outline/section-card/SectionCard.jsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+
+import { getSectionStatus } from '../utils';
+import CardHeader from '../card-header/CardHeader';
```
Comment (PKulkoRaccoonGang) [inline src/course-outline/section-card/SectionCard.jsx:5]:
```suggestion
import CardHeader from '../card-header/CardHeader';
import { getSectionStatus } from '../utils';
```

**Review by PKulkoRaccoonGang — COMMENTED** [review]
Please add developers who should review through the GitHub interface (they will receive a notification in the form of an email and pay attention)

image

File: src/course-outline/card-header/CardHeader.jsx (hunk 1 above)
Comment (vladislavkeblysh) [reply src/course-outline/card-header/CardHeader.jsx]: Done

File: src/course-outline/card-header/messages.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    id: 'course-authoring.course-outline.section.menu.duplicate',
+    defaultMessage: 'Duplicate',
+  },
+  menuDelete: {
+    id: 'course-authoring.course-outline.section.menu.delete',
+    defaultMessage: 'Delete',
+  },
+  menuNewSubsection: {
+    id: 'course-authoring.course-outline.section.menu.new-subsection',
+    defaultMessage: 'New Subsection',
```
[the remark this answers is not in the document]
Comment (vladislavkeblysh) [reply src/course-outline/card-header/messages.js]: Done

File: src/course-outline/section-card/SectionCard.jsx (hunk 2 above)
Comment (vladislavkeblysh) [reply src/course-outline/section-card/SectionCard.jsx]: Done

File: src/course-outline/section-card/SectionCard.jsx [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+      )}
+    </div>
+  );
+};
+
+SectionCard.defaultProps = {
+  children: null,
+};
+
+// TODO: add new props
```
[the remark this answers is not in the document]
Comment (vladislavkeblysh) [reply src/course-outline/section-card/SectionCard.jsx]: Done

File: src/i18n/messages/ar.json [hunk 5]
```diff
@@ -746,6 +746,18 @@
...
+  "course-authoring.course-outline.section.status-badge.live": "Live",
+  "course-authoring.course-outline.section.status-badge.published-not-live": "Published not live",
+  "course-authoring.course-outline.section.status-badge.staff-only": "Staff only",
+  "course-authoring.course-outline.section.status-badge.draft": "Draft",
+  "course-authoring.course-outline.section.menu.edit": "Edit",
+  "course-authoring.course-outline.section.menu.publish": "Publish",
+  "course-authoring.course-outline.section.menu.configure": "Configure",
+  "course-authoring.course-outline.section.menu.duplicate": "Duplicate",
+  "course-authoring.course-outline.section.menu.delete": "Delete",
+  "course-authoring.course-outline.section.menu.new-subsection": "New Subsection",
```
[the remark this answers is not in the document]
Comment (vladislavkeblysh) [reply src/i18n/messages/ar.json]: Done

File: src/course-outline/section-card/SectionCard.jsx [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+  return (
+    <div className="section-card">
+      <CardHeader
+        title={displayName}
+        sectionStatus={sectionStatus}
+        isExpanded={isExpanded}
+        handleExpand={handleExpandContent}
+      />
+      {isExpanded && (
```
Comment (ruzniaievdm) [inline src/course-outline/section-card/SectionCard.jsx:39]: you can use collapsible component from the paragon here
https://paragon-openedx.netlify.app/components/collapsible/

Comment (ruzniaievdm) [reply src/course-outline/section-card/SectionCard.jsx:39]: as i understand we planned to show section highlights even isExpanded false. As i was was on the legacy

File: src/course-outline/constants.js [hunk 7]
```diff
@@ -1,3 +1,12 @@
+export const SECTION_BADGE_STATUTES = {
+  live: 'live',
+  published_not_live: 'published_not_live',
```
Comment (ruzniaievdm) [inline src/course-outline/constants.js:3]: change keys to camelCase

**Review by ruzniaievdm — CHANGES_REQUESTED** [review]

**Review by PKulkoRaccoonGang — APPROVED** [review]

File: src/course-outline/constants.js (hunk 7 above)
Comment (vladislavkeblysh) [reply src/course-outline/constants.js]: Done

File: src/course-outline/section-card/SectionCard.jsx (hunk 6 above)
Comment (vladislavkeblysh) [reply src/course-outline/section-card/SectionCard.jsx]: Done

File: src/course-outline/card-header/CardHeader.jsx [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+          alt="section-card-header__menu"
+          iconAs={Icon}
+        />
+        <Dropdown.Menu>
+          <Dropdown.Item>{intl.formatMessage(messages.menuEdit)}</Dropdown.Item>
+          <Dropdown.Item>{intl.formatMessage(messages.menuPublish)}</Dropdown.Item>
+          <Dropdown.Item>{intl.formatMessage(messages.menuConfigure)}</Dropdown.Item>
+          <Dropdown.Item>{intl.formatMessage(messages.menuDuplicate)}</Dropdown.Item>
+          <Dropdown.Item>{intl.formatMessage(messages.menuDelete)}</Dropdown.Item>
+          <Dropdown.Item>{intl.formatMessage(messages.menuNewSubsection)}</Dropdown.Item>
```
Comment (wowkalucky) [inline src/course-outline/card-header/CardHeader.jsx:83]: This item was declined yesterday...

File: src/course-outline/card-header/CardHeader.jsx [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+        <Dropdown.Toggle
+          className="section-card-header__menu"
+          id="section-card-header__menu"
+          as={IconButton}
+          src={MoveVertIcon}
+          alt="section-card-header__menu"
+          iconAs={Icon}
+        />
+        <Dropdown.Menu>
+          <Dropdown.Item>{intl.formatMessage(messages.menuEdit)}</Dropdown.Item>
```
Comment (wowkalucky) [inline src/course-outline/card-header/CardHeader.jsx:78]: This action is in discussion now.

File: src/course-outline/card-header/CardHeader.jsx (hunk 8 above)
Comment (vladislavkeblysh) [reply src/course-outline/card-header/CardHeader.jsx]: Done

**Pull request merged**